genesis/src/lib.rs

```diff
@@ -213,12 +212,23 @@ impl RawGenesisBlockBuilder {
             transaction: GenesisTransaction {
                 isi: SmallVec::new(),
             },
+            state: validator_state::Unset,
         }
     }
 
+    /// Set the validator.
+    pub fn validator(self, validator: Validator) -> RawGenesisBlockBuilder<validator_state::Set> {
+        RawGenesisBlockBuilder {
+            transaction: self.transaction,
+            state: validator_state::Set(validator),
+        }
+    }
+}
+
+impl<S> RawGenesisBlockBuilder<S> {
```

Comment:
You can combine this impl with the same impl below.

Reply:
I think you confused `RawGenesisBlockBuilder` with `RawGenesisDomainBuilder`

Reply:
oh, right